pandas/core/groupby/generic.py

```diff
@@ -1569,6 +1574,198 @@ def func(df):
+                medium     FR         0.25
+        Name: count, dtype: float64
+        """
+        RESULT_NAME = "count"
+        if self.axis == 1:
+            raise NotImplementedError(
+                "DataFrameGroupBy.value_counts only handles axis=0"
+            )
+
+        with self._group_selection_context():
+            df = self.obj
+
+            # Check for mapping, array or function rather than column grouping
+            non_column_grouping = not self.grouper.groupings[0].in_axis
+
+            # Try to find column names
+            if non_column_grouping:
+                keys: Any = []
+                remaining_columns = self._selected_obj.columns
+            elif isinstance(self._selected_obj, Series):
+                keys = [grouping.name for grouping in self.grouper.groupings]
+                remaining_columns = [self._selected_obj.name]
+            else:
+                if isinstance(self.keys, ops.BaseGrouper):
+                    keys = [grouping.name for grouping in self.keys.groupings]
+                elif isinstance(self.keys, str):
+                    keys = [self.keys]
+                else:
+                    keys = self.keys
+
+                remaining_columns = [
+                    key for key in self._selected_obj.columns if key not in keys
+                ]
+
+            if subset is not None:
+                for key in subset:
+                    if key in keys:
+                        raise ValueError(
+                            f"Key {key} in subset cannot be one of "
+                            "the groupby column keys"
+                        )
+                remaining_columns = subset
+
+            if dropna:
+                df = df.dropna(subset=remaining_columns, axis="index", how="any")
+
+            # Add the remaining_column keys to the main grouper
+            main_grouper, _, _ = get_grouper(
+                df,
+                key=self.keys,
+                axis=self.axis,
+                level=self.level,
+                sort=self.sort,
+                mutated=self.mutated,
+                dropna=self.dropna,
+            )
+            groupings = list(main_grouper.groupings)
+            for key in remaining_columns:
+                grouper, _, _ = get_grouper(
+                    df,
+                    key=key,
+                    axis=self.axis,
+                    level=self.level,
+                    sort=self.sort,
+                    mutated=self.mutated,
+                    # nulls have already been dropped from remaining_columns
+                    dropna=False,
+                )
+                groupings += list(grouper.groupings)
+
+            # Take the size of the overall columns
+            result = df.groupby(
+                groupings,
+                sort=self.sort,
+                observed=self.observed,
+                dropna=self.dropna,
+            ).size()
+            result.name = RESULT_NAME
```

Comment:
Other groupby ops don't set name on the resulting Series when as_index=True, but will name the column when as_index=False. I think we should follow that pattern here.

Reply:
Done